tests/components/openweathermap/test_config_flow.py

```diff
@@ -28,190 +23,155 @@
 
-        conf_entries = hass.config_entries.async_entries(DOMAIN)
-        entry = conf_entries[0]
-        assert entry.state is ConfigEntryState.LOADED
+@patch("homeassistant.components.openweathermap.config_flow.OWMClient")
+@patch("homeassistant.components.openweathermap.OWMClient")
+async def test_form(owm_client_mock, owm_client_mock2, hass: HomeAssistant) -> None:
+    """Test that the form is served with valid input."""
+    mock = _create_mocked_owm_client(True)
+    owm_client_mock.return_value = mock
+    owm_client_mock2.return_value = mock
 
-        await hass.config_entries.async_unload(conf_entries[0].entry_id)
-        await hass.async_block_till_done()
-        assert entry.state is ConfigEntryState.NOT_LOADED
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
 
-        assert result["type"] is FlowResultType.CREATE_ENTRY
-        assert result["title"] == CONFIG[CONF_NAME]
-        assert result["data"][CONF_LATITUDE] == CONFIG[CONF_LATITUDE]
-        assert result["data"][CONF_LONGITUDE] == CONFIG[CONF_LONGITUDE]
-        assert result["data"][CONF_API_KEY] == CONFIG[CONF_API_KEY]
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
 
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
+    )
 
-async def test_form_options(hass: HomeAssistant) -> None:
-    """Test that the options form."""
-    mocked_owm = _create_mocked_owm(True)
+    await hass.async_block_till_done()
 
-    with patch(
-        "pyowm.weatherapi25.weather_manager.WeatherManager",
-        return_value=mocked_owm,
-    ):
-        config_entry = MockConfigEntry(
-            domain=DOMAIN, unique_id="openweathermap_unique_id", data=CONFIG
-        )
-        config_entry.add_to_hass(hass)
+    conf_entries = hass.config_entries.async_entries(DOMAIN)
+    entry = conf_entries[0]
+    assert entry.state is ConfigEntryState.LOADED
 
-        assert await hass.config_entries.async_setup(config_entry.entry_id)
-        await hass.async_block_till_done()
+    await hass.config_entries.async_unload(conf_entries[0].entry_id)
+    await hass.async_block_till_done()
+    assert entry.state is ConfigEntryState.NOT_LOADED
 
-        assert config_entry.state is ConfigEntryState.LOADED
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == CONFIG[CONF_NAME]
+    assert result["data"][CONF_LATITUDE] == CONFIG[CONF_LATITUDE]
+    assert result["data"][CONF_LONGITUDE] == CONFIG[CONF_LONGITUDE]
+    assert result["data"][CONF_API_KEY] == CONFIG[CONF_API_KEY]
 
-        result = await hass.config_entries.options.async_init(config_entry.entry_id)
 
-        assert result["type"] is FlowResultType.FORM
-        assert result["step_id"] == "init"
+@patch("homeassistant.components.openweathermap.config_flow.OWMClient")
```

Comment:
It probably makes sense to make these pytest fixtures

Reply:
updated